File: src/lua/dynamic_array_with_builtin.lua
itemCount = tonumber(arg[1],10)
print(string.format("Testing a push/pop of %d builtin items.", itemCount))
table = {}
evenNumberCount = 0
for arrayVal = 0,itemCount do
    table[arrayVal] = arrayVal
end

for arrayVal = 0,itemCount do
    if table[arrayVal]%2==0
    then
        evenNumberCount = evenNumberCount + 1
    end
    table[arrayVal] = nil
end

print(string.format("There were %d even numbers.", evenNumberCount))

File: src/go/dynamic_array_with_builtin/dynamic_array_with_builtin.go
package main

import (
	"fmt"
	"os"
	"strconv"
)

func main() {
	unparsedPushedItems := os.Args[1]
	pushedItems, _ := strconv.Atoi(unparsedPushedItems)
	fmt.Printf("Testing a push/pop with %v builtin items.\n", pushedItems)

	var myslc []int
	var evenNumberCount = 0
	for i := range pushedItems {
		myslc = append(myslc, i)
	}
	for i := range pushedItems {
		popInt := myslc[i]
		if popInt%2 == 0 {
			evenNumberCount += 1
		}
	}
	fmt.Printf("There were %v even numbers.\n", evenNumberCount)
}
